tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -26,6 +26,13 @@
  */
 class ValidatorTest extends TestCase
 {
+    /**
+     * Used to flag nested rules created with addNested() and addNestedMany()
+     *
+     * @var string
+     */
+    protected const NESTED = '_nested';
```

Comment:
Given that you need this here, this probably shouldn't be a protected constant on the class.

Reply:
Given that the outside  (and as such the test) isnt supposed to use it directly :)
Not sure...

Reply:
Could the tests be changed to not need 'internal' parts of the class? If not those identifiers are probably not internal.

Reply:
It just looked a bit like protected methods (which with a trick you can also separately invoke/test).
If you need to access, using magic string in tests or a cloned constant sounds fine.
But for simplicity we can sure keep it as part of public API and to more easily access from tests etc.
Fine with me - I can revert this.

Reply:
I think we can start using protected constants more in the future.